src/shared/copy.c

```diff
@@ -794,6 +797,53 @@ static int prepare_nocow(int fdf, const char *from, int fdt, unsigned *chattr_ma
         return 0;
 }
 
+/* Copies fs-verity status.  May re-open fdt to do its job. */
+static int copy_fs_verity(int fdf, int *fdt) {
+#if HAVE_LINUX_FSVERITY_H
+        struct fsverity_descriptor desc;
+        struct fsverity_read_metadata_arg read_arg = {
+                .metadata_type = FS_VERITY_METADATA_TYPE_DESCRIPTOR,
+                .buf_ptr = (uint64_t) &desc,
+                .length = sizeof desc,
+        };
+
+        int ret = ioctl(fdf, FS_IOC_READ_VERITY_METADATA, &read_arg);
+        if (ret < 0) {
+                /* These mean that fs-verity isn't enabled on the file, filesystem, or kernel.  In all of
+                 * those cases, the correct thing to do is to do nothing at all.
+                 */
+                if (errno == ENODATA || errno == ENOTTY || errno == ENOTSUP)
+                        return 0;
+                return -errno;
+        }
+
+        if (ret != sizeof desc) {
```

Comment:
The struct is versioned, so it's not inconceivable that it could change in the future. But it won't shrink, it will only grow for backward compat, hence error out if it's smaller than expected instead. Presumably any future new parameter can just be ignored